### PR TITLE
Ready for reviews - GUVNOR-3193 Change Project from a folder in a repo to a repository

### PR DESCRIPTION
**Pull request by Rikkola**
Please do not merge.

About 3 lines of actual changes. The renames and looks like code formatting moved methods from one location to another.

https://issues.jboss.org/browse/AF-648

The related PRs:
- https://github.com/kiegroup/kie-soup/pull/7
- https://github.com/kiegroup/appformer/pull/104
- https://github.com/kiegroup/drools/pull/1402
- https://github.com/kiegroup/kie-wb-common/pull/1015
- https://github.com/kiegroup/drools-wb/pull/566
- https://github.com/kiegroup/jbpm-designer/pull/643
- https://github.com/kiegroup/jbpm-wb/pull/832
- https://github.com/kiegroup/optaplanner-wb/pull/196
- https://github.com/kiegroup/kie-wb-distributions/pull/573
- https://github.com/kiegroup/kie-docs/pull/558

### CODE REVIEW

File: jbpm-designer-backend/src/main/java/org/jbpm/designer/web/preprocessing/impl/JbpmPreprocessingUnit.java [hunk 1]
```diff
@@ -144,32 +150,117 @@
...
     @Inject
-    KieProjectService kieProjectService;
-
-    @Inject
     private LRUBuilderCache builderCache;
 
     public JbpmPreprocessingUnit() {
     }
 
+    protected static String readFile(String pathname) throws IOException {
```
Comment (Rikkola) [inline jbpm-designer-backend/src/main/java/org/jbpm/designer/web/preprocessing/impl/JbpmPreprocessingUnit.java]: Looks like a  bad merge. Need to clean this up.

Comment (Rikkola) [reply jbpm-designer-backend/src/main/java/org/jbpm/designer/web/preprocessing/impl/JbpmPreprocessingUnit.java]: Nope. Just for some reason my code formatting moved these.

File: jbpm-designer-backend/src/test/java/org/jbpm/designer/web/preprocessing/impl/JbpmPreprocessingUnitVFSGitTest.java [hunk 2]
```diff
@@ -654,9 +655,11 @@ private String decodeUniqueId(String uniqueId) {
...
     }
 
-    private LRUBuilderCache getBuilderCache(KieProject kieProject) {
-        NewProjectEvent event = mock(NewProjectEvent.class);
-        when(event.getProject()).thenReturn(kieProject);
+    private LRUBuilderCache getBuilderCache(KieModule kieModule) {
+        final NewProjectEvent event = mock(NewProjectEvent.class);
+        final Project project = mock(Project.class);
+        when(event.getProject()).thenReturn(project);
+        when(project.getMainModule()).thenReturn(kieModule);
```
Comment (Rikkola) [inline jbpm-designer-backend/src/test/java/org/jbpm/designer/web/preprocessing/impl/JbpmPreprocessingUnitVFSGitTest.java]: Small change here.

**Comment by tsurdilo** [comment]
Jenkins retest please

**Comment by Rikkola** [comment]
@tsurdilo Working on this at the moment. There are some broken unit tests and trying to figure out how I broke them.

**Comment by Rikkola** [comment]
@tsurdilo Also, you should get some sleep. It is night there.

**Comment by tsurdilo** [comment]
@Rikkola ok now to merge?

**Pull request closed without merging**